soufi/finder.py

```diff
@@ -172,12 +178,27 @@ def distro(self) -> Union[Distro, str]:
         raise NotImplementedError  # pragma: no cover
 
     def __init__(
-        self, name: str = None, version: str = None, s_type: SourceType = None
+        self,
+        name: str = None,
+        version: str = None,
+        s_type: SourceType = None,
+        cache_backend: str = None,
+        cache_args: Iterable[str] = None,
     ):
         self.name = name
         self.version = version
         self.s_type = s_type
 
+        # By default, use a small in-memory LRU cache, 512 objects large
+        if not cache_backend:
+            cache_backend = 'dogpile.cache.memory'
```

Comment:
This should be null by default, not memory.

Reply:
That's fair; I was aiming for a drop-in replacement for `functools.lru_cache`, but this would make the implementation simpler.

Reply:
I would also say, we could consider passing `dogpile.cache.memory` from the CLI to keep the current behaviour on there.

Reply:
Absolutely.